stdlib/src/memory/boxed.mojo

```diff
@@ -0,0 +1,143 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+from memory import UnsafePointer, stack_allocation, memcpy
+
+
+struct Boxed[T: AnyType, address_space: AddressSpace = AddressSpace.GENERIC]:
```

Comment:
**Suggestion** As mentioned as a point in the summarized review: let's drop this `address_space` parameter for now and just fix the held data (`_inner`) to use `AddressSpace.GENERIC` for now.  We can evaluate this lazily/add it when we have a clear use case for other address spaces.  You really want finer control over the allocations/deallocations, which this address space — in its current form, won't help you with.  We should wait until we have a proper design for custom allocators.